### PR TITLE
Use `with open` in `scapy/data.py`

### PR DESCRIPTION
**Pull request by carlbordum**

### CODE REVIEW

File: scapy/data.py [hunk 1]
```diff
@@ -126,22 +126,21 @@ def load_protocols(filename, _integer_base=10):
...
+                        line = line[:shrp]
+                    line = line.strip()
+                    if not line:
+                        continue
+                    lt = tuple(re.split(spaces, line))
+                    if len(lt) < 2 or not lt[0]:
+                        continue
+                    dct[lt[0]] = int(lt[1], _integer_base)
+                except Exception as e:
+                    log_loading.info("Couldn't parse file [%s]: line [%r] (%s)", filename, line, e)  # noqa: E501
```
Comment (gpotter2) [inline scapy/data.py]: Hi ! Thanks for the PR.

We’re trying to ensure that everyone who edits a line fixes its #noqa tag (if possible), to progressively improve the code base (the noqa were automatic).

Could you please update this line too and fix it (this noqa means line too long > 79 characters) ? Thanks !

File: scapy/data.py [hunk 2]
```diff
@@ -126,22 +126,21 @@ def load_protocols(filename, _integer_base=10):
...
-                if not line:
-                    continue
-                lt = tuple(re.split(spaces, line))
-                if len(lt) < 2 or not lt[0]:
-                    continue
-                dct[lt[0]] = int(lt[1], _integer_base)
-            except Exception as e:
-                log_loading.info("Couldn't parse file [%s]: line [%r] (%s)", filename, line, e)  # noqa: E501
-        f.close()
+        with open(filename, "rb") as f:
```
Comment (guedou) [inline scapy/data.py]: While you are modifying this code, could you also rename `f` to `fdesc` ? Thanks.

**Comment by carlbordum** [comment]
Alright, I did what you asked for!

Can you tell Travis to rebuild? there was an unrelated error.

**Comment by p-l-** [comment]
@carlbordum done.

**Comment by carlbordum** [comment]
Thanks.

I don't understand why codecov decreased. Can you help me out here?

**Pull request merged**